Client/Interface/I3dEntities.hpp

```diff
@@ -0,0 +1,88 @@
+/*
+** EPITECH PROJECT, 2024
+** IGraphic.hpp
+** File description:
+** DESCRIPTION
+*/
+
+#pragma once
+
+#include <vector>
+#include <string>
+
+#include "IColor.hpp"
+
+/**
+ * @class I3DEntity
+ * @brief Interface for 3D entity operations.
+ *
+ * This interface defines methods for loading 3D entities, textures, and setting various properties such as position, rotation, scale, and color.
+ */
+class I3DEntity {
+    public:
+        /**
+         * @enum MATERIAL
+         * @brief Enumeration of material types for 3D entities.
+         */
+        enum MATERIAL {
+            MATERIAL_MAP_ALBEDO,       ///< Albedo map
+            MATERIAL_MAP_METALNESS,    ///< Metalness map
+            MATERIAL_MAP_NORMAL,       ///< Normal map
+            MATERIAL_MAP_ROUGHNESS,    ///< Roughness map
+            MATERIAL_MAP_OCCLUSION,    ///< Occlusion map
+            MATERIAL_MAP_EMISSION,     ///< Emission map
+            MATERIAL_MAP_HEIGHT,       ///< Height map
+            MATERIAL_MAP_CUBEMAP,      ///< Cubemap
+            MATERIAL_MAP_IRRADIANCE,   ///< Irradiance map
+            MATERIAL_MAP_PREFILTER,    ///< Prefilter map
+            MATERIAL_MAP_BRDF          ///< BRDF map
+        };
+
+        /**
+         * @brief Virtual destructor for I3DEntity.
+         */
+        virtual ~I3DEntity() = default;
+
+        /**
+         * @brief Load a 3D entity from a file.
+         * @param path The path to the file containing the 3D entity.
+         * @return The loaded 3D entity.
+         */
+        virtual I3DEntity Load3DEntity(std::string const &path) = 0;
+
+        /**
+         * @brief Load textures for the 3D entity.
+         * @param textures A vector of pairs containing texture file paths and their corresponding material types.
+         */
+        virtual void LoadTexture(std::vector<std::pair<std::string, MATERIAL>>) = 0;
```

Comment:
The name of the parameter is missing, and you should rename it `LoadTextures` coz as the parameter tells, you can load severaltextures